File: 0100-same-tree/0100-same-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean isSameTree(TreeNode p, TreeNode q) {
        Queue<TreeNode> qu = new LinkedList<>();
        qu.add(p);
        qu.add(q);
        while (!qu.isEmpty()) {
            TreeNode n1 = qu.poll();
            TreeNode n2 = qu.poll();
            if (n1 == null && n2 == null) {
                continue;
            }
            if (n1 == null || n2 == null || n1.val != n2.val) {
                return false;
            }
            qu.add(n1.left);
            qu.add(n2.left);
            qu.add(n1.right);
            qu.add(n2.right);
        }  
        return true;
    }
}

File: 0100-same-tree/0100-same-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        stack = [p, q]
        while stack:
            nl = stack.pop()
            nr = stack.pop()
            if not nl and not nr:
                continue
            if not nl or not nr or nl.val != nr.val:
                return False
            stack.append(nl.left)
            stack.append(nr.left)
            stack.append(nl.right)
            stack.append(nr.right)
        return True